### ISSUE TITLE
The IDENTIFY command sent to nickserv  should be configurable.

### ISSUE THREAD

**Issue by sitedyno** [phergie/phergie-irc-plugin-react-nickserv#12]
Phergie currently uses `:IDENTIFY <nick> <password>` to identify to nickserv. Which means the bot will try to identify with its current nick. As you can see below having the bot's nick in the identify command is for identifying with a **different** nick than the one the bot is currently using. On top of that Rizon does not account for identifying for a different nick than the current nick and all IDENTIFY commands fail on Rizon. I'm sure there are other networks that expect such behavior.

Since `:IDENTIFY <password>` is the common denominator Phergie should use that format or allow the format to be specified by the user.

Espernet:
>NickServ: ***** NickServ Help *****
>NickServ: Help for IDENTIFY:
>NickServ:
>NickServ: IDENTIFY identifies you with services so that you
>NickServ: can perform general maintenance and commands that
>NickServ: require you to be logged in.
>NickServ:
>NickServ: Syntax: `IDENTIFY <password>`
>NickServ:
>NickServ: You can also identify for another nick than you
>NickServ: are currently using.
>NickServ:
>NickServ: Syntax: `IDENTIFY <nick> <password>`
>NickServ:
>NickServ: Example:
>NickServ:     /msg NickServ IDENTIFY foo
>NickServ:     /msg NickServ IDENTIFY jilles foo
>NickServ: ***** End of Help *****

Freenode:
>NickServ: ***** NickServ Help *****
>NickServ: Help for IDENTIFY:
>NickServ:
>NickServ: IDENTIFY identifies you with services so that you
>NickServ: can perform general maintenance and commands that
>NickServ: require you to be logged in.
>NickServ:
>NickServ: Syntax: `IDENTIFY <password>`
>NickServ:
>NickServ: You can also identify for another nick than you
>NickServ: are currently using.
>NickServ:
>NickServ: Syntax: `IDENTIFY <nick> <password>`
>NickServ:
>NickServ: Example:
>NickServ:     /msg NickServ IDENTIFY foo
>NickServ:     /msg NickServ IDENTIFY jilles foo
>NickServ: ***** End of Help *****

Rizon:
>help identify
>NickServ: Syntax: `IDENTIFY password`
>NickServ:
>NickServ: Tells NickServ that you are really the owner of this
>NickServ: nick.  Many commands require you to authenticate yourself
>NickServ: with this command before you use them.  The password
>NickServ: should be the same one you sent with the REGISTER
>NickServ: command.

**Comment by Renegade334** [comment]
Peut-être?

``` diff
diff --git a/src/Plugin.php b/src/Plugin.php
index 728857c..93d77fc 100644
--- a/src/Plugin.php
+++ b/src/Plugin.php
@@ -32,6 +32,17 @@ class Plugin extends AbstractPlugin
     protected $password;
 
     /**
+     * Identification command pattern.
+     *
+     * String patterns:
+     * - %nickname% - bot's current nickname
+     * - %password% - NickServ password
+     *
+     * @var string
+     */
+    protected $identifyCommand = 'IDENTIFY %nickname% %password%';
+
+    /**
      * Regex pattern matching a request for identification
      *
      * @var string
@@ -107,6 +118,9 @@ class Plugin extends AbstractPlugin
         if (!empty($config['ghost'])) {
             $this->ghostEnabled = true;
         }
+        if (isset($config['identifycommand'])) {
+            $this->identifyCommand = $this->getConfigOption($config, 'identifycommand');
+        }
         if (isset($config['identifypattern'])) {
             $this->identifyPattern = $this->getConfigOption($config, 'identifypattern');
         }
@@ -156,7 +170,12 @@ class Plugin extends AbstractPlugin
 
         // Authenticate the bot's identity for authentication requests
         if (preg_match($this->identifyPattern, $message)) {
-            return $queue->ircPrivmsg($this->botNick, 'IDENTIFY ' . $connection->getNickname() . ' ' . $this->password);
+            $command = str_replace(
+                ['%nickname%', '%password%'],
+                [$connection->getNickname(), $this->password],
+                $this->identifyCommand
+            );
+            return $queue->ircPrivmsg($this->botNick, $command);
         }
 
         // Emit an event on successful authentication

```

**Comment by elazar** [comment]
That would work. I would make a point of documenting it fairly clearly in the README, including mention of the fact the pattern needed for it to work on Rizon.

**Comment by elazar** [comment]
Merged #14.

**Comment by sitedyno** [comment]
Thanks @Renegade334 :)
